src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java

```diff
@@ -96,20 +92,19 @@
 		Controller controller;
 
 		@Override
-		public final StreamCreateFunction apply(final String scopeName)
-		throws ScopeCreateException {
+		public final StreamCreateFunction apply(final String scopeName) {
 			try {
-				if(controller.createScope(scopeName).get(CONTROL_API_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS)) {
+				if(controller.createScope(scopeName).get(controlApiTimeoutMillis, TimeUnit.MILLISECONDS)) {
 					Loggers.MSG.trace("Scope \"{}\" was created", scopeName);
 				} else {
 					Loggers.MSG.info("Scope \"{}\" was not created, may be already existing before", scopeName);
 				}
-				return new StreamCreateFunctionImpl(controller, scopeName);
 			} catch(final InterruptedException e) {
 				throw new InterruptRunException(e);
 			} catch(final Throwable cause) {
-				throw new ScopeCreateException(scopeName, cause);
+				LogUtil.exception(Level.WARN, cause, "{}: failed to create the scope \"{}\"", stepId, scopeName);
```

Comment:
Could you please explain this change?

Reply:
Sure. We shouldn't fail the load operation on this stage because the failure may be spurious. The scope may be existing, but the controller fails the call with something like TimeoutException. And the following load operation executes successfully.